mindsdb/interfaces/agents/langchain_agent.py

```diff
@@ -435,8 +436,9 @@ def create_agent(self, df: pd.DataFrame, args: Dict = None) -> AgentExecutor:
         # Set up tools.
         llm = create_chat_model(args)
         self.llm = llm
+
+        # Don't set embedding model for retrieval mode - let the knowledge base handle it
         if args.get("mode") == "retrieval":
-            self.set_embedding_model(args)
```

Comment:
this was causing issue, we move the setting of embeddings model to retrieval_tool